zeroconf/src/linux/entry_group.rs

```diff
@@ -89,8 +89,42 @@ impl ManagedAvahiEntryGroup {
                 )
             },
             "could not register service",
-        )?;
+        )
+    }
+
+    /// Delegate function for [`avahi_entry_group_add_service_subtype()`].
+    ///
+    /// Also propagates any error returned into a `Result`.
+    ///
+    /// [`avahi_entry_group_add_service_subtype()`]: https://avahi.org/doxygen/html/publish_8h.html#a93841be69a152d3134b408c25bb4d5d5
+    pub fn add_service_subtype(
```

Comment:
Could you please add a unit test for this function in a `tests` module in this file? I know we don't have great coverage currently but I would like to start moving in the direction of better test coverage.

Reply:
I ended up adding it to the existing `service_test.rs` test with a loop to browse for the service with and without the subtype. Additionally I added a timeout to this test since I noticed that it's failure mode was to run forever.